chapter01/chapter01.adoc

```diff
@@ -0,0 +1,263 @@
+
+[[glossary]]
+== Glossary
+
+[[microservices]]
+Microservices:: An architectural style for building applications as a collection of small, independent services. Each service focuses on a specific business capability and communicates with other services through well-defined APIs.
+
+[[apis]]
+APIs (Application Programming Interfaces):: A set of definitions and protocols that specify how software components interact with each other. 
+
+[[cloud-native-development]]
+Cloud-native development:: An approach to building and running applications that are specifically designed for the cloud environment. It involves using technologies and practices that leverage the benefits of cloud platforms, such as scalability, elasticity, and pay-as-you-go pricing.
+
+[[eclipse-foundation-working-group]]
+Eclipse Foundation Working Group:: A collaborative group of industry leaders and Java community members who actively contribute to the of development of Eclipse projects like MicroProfile within the Eclipse Foundation framework.
+
+[[jakarta-ee]]
+Jakarta EE:: Jakarta EE (formerly Java Platform, Enterprise Edition, or Java EE) is a set of specifications, extending Java Platform, Standard Edition, or Java SE with specifications for enterprise features such as web services, database persistence, asynchronous messaging and more.
+
+[[external-configuration]]
+External Configuration:: A technique in application development where configuration data is separated from the application code, allowing the application's behavior to be adjusted without changing the code, especially useful in cloud-native and microservices architectures.
+
+[[health-checks]]
+Health Checks:: Mechanisms used in microservices architectures to continuously check the status of an application or service to ensure it is functioning correctly and available to users.
+
+[[fault-tolerance]]
+Fault Tolerance:: The ability of a system to continue operating in the event of the failure of some of its components. This feature is critical for maintaining high availability and reliability in microservices architectures.
+
+[[vendor-neutrality]]
+Vendor Neutrality:: The principle of designing software products and standards not controlled by any single vendor, promoting user interoperability and choice.
+
+[[interoperability]]
+Interoperability:: The ability of a software to exchange and make use of information across different platforms and services.
+
```

Comment:
Not sure if it is a goal to include tutorials for the Jakarta EE components. It might be worthwhile to reference Jakarta EE tutorials for basics and only cover additional details that pertain to integration with these components in MicroProfile. The Jakarta EE components are included are json-p, json-b, jax-rs (please use the current name: Jakarta RESTful Web Services (or REST)), cdi. Core profile also includes interceptors and annotations but I'm not sure that needs mention in the introduction. It might be enough in this introduction to simply mention these are jakarta ee specifications and only because the bulk of the definition for these is likely found there.

Reply:
At the time MicroProfile started, it was called Java EE, JSON-P, JSON-B, and JAX-RS, that's why using it only here. In rest of the tutorial would be using current names.